### DISCUSSION TITLE
Wallet connection on zkSync plugin

### DISCUSSION THREAD

**Discussion by mehdi-ceng** [Cyfrin/foundry-full-course-cu#2680] [Q&A] [answered]
I'm at  Lesson 16, Section1 of Solidity Smart Contract Development course - zkSync Plugin - and I try to connect wallet via metamask, but metamask extension does not show the interface that lets me deploy my smart contract. Also there is small blue square to up right of the metamask icon that shows number of times I repeated that step. Any ideas what is happening?

**Comment by EngrPips** [comment]
Hello @mehdi-ceng, refresh your browser and use the `wallet connect` option instead of `injected provider` option, and let's know how it plays out.

**Reply by mehdi-ceng** [reply, marked as answer]
In zkSync plugin, there is no `inject provider` and instead there are four option in top most drop down button in plugin UI as of writing this post(which consists of two drop downs: Environment and Manual Accounts): (in Environment) `wallet, local devnet, remote devnet, custom network`. After refreshing browser and choosing `wallet` option, I was able to connect my zkSync account(on MetaMask) to Remix IDE and deploy the contract. Thanks for help.

**Reply by EngrPips** [reply]
You are welcome, Keep crushing it.
